platform/xbox/include/math.h

```diff
@@ -51,11 +52,69 @@ typedef double double_t;
     #define M_SQRT1_2 0.70710678118654752440
 #endif
 
-#define fpclassify(x) (__builtin_fpclassify(FP_NAN, FP_INFINITE, FP_NORMAL, FP_SUBNORMAL, FP_ZERO, x))
-#define isinf(x) (__builtin_isinf_sign(x))
-#define isnan(x) (__builtin_isnan(x))
-#define isnormal(x) (__builtin_isnormal(x))
-#define signbit(x) (__builtin_signbit(x))
+static inline int __fp_unordered_compare (long double x, long double y)
+{
+    unsigned short sw;
+    __asm__ ("fucom;"  // compare st(0) with st(1)
+             "fnstsw;" // store status word in ax (sw variable)
+             : "=a"(sw) : "t"(x), "u"(y));
+    return sw;
+}
+
+int __fpclassifyf (float x);
+int __fpclassify (double x);
+int __fpclassifyl (long double x);
+int __signbitf (float x);
+int __signbit (double x);
+int __signbitl (long double x);
+
+#ifndef __cplusplus
+    #define fpclassify(x) (sizeof(x) == sizeof(float) ? __fpclassifyf(x) : sizeof(x) == sizeof(double) ? __fpclassify(x) : __fpclassifyl(x))
+    #define isfinite(x) ((fpclassify(x) & FP_NAN) == 0)
+    #define isinf(x) (fpclassify(x) == FP_INFINITE)
+    #define isnan(x) (fpclassify(x) == FP_NAN)
+    #define isnormal(x) (fpclassify(x) == FP_NORMAL)
+    #define signbit(x) (sizeof(x) == sizeof(float) ? __signbitf(x) : sizeof(x) == sizeof(double) ? __signbit(x) : __signbitl(x))
+
+    // 7.12.14
+    #define isgreater(x, y) ((__fp_unordered_compare((x), (y)) & 0x4500) == 0)
+    #define isgreaterequal(x, y) ((__fp_unordered_compare((x), (y)) & FP_INFINITE) == 0)
+    #define isless(x, y) ((__fp_unordered_compare((y), (x)) & 0x4500) == 0)
+    #define islessequal(x, y) ((__fp_unordered_compare((y), (x)) & FP_INFINITE) == 0)
+    #define islessgreater(x, y) ((__fp_unordered_compare((x), (y)) & FP_SUBNORMAL) == 0)
+    #define isunordered(x, y) ((__fp_unordered_compare((x), (y)) & 0x4500) == 0x4500)
+#else
+    extern "C++"
+    {
+        inline int fpclassify (float x) { return __fpclassifyf(x); };
+        inline int fpclassify (double x) { return __fpclassify(x); };
+        inline int fpclassify (long double x) { return __fpclassifyl(x); };
+        template <typename T>
+        inline int isfinite (T x) { return #define isfinite(x) ((fpclassify(x) & FP_NAN) == 0); };
```

Comment:
This was a syntax error (probably caused by search & replace) which slipped through review.
Please send a PR to fix this!